sid/contacts.py

```diff
@@ -208,7 +204,7 @@ def _calculate_infections_numba(
             n_contacts = contacts[i, cm]
             for _ in range(n_contacts):
                 contact_takes_place = True
-                group_j = _choose_one_element(groups_list[cm], weights=group_probs_i)
+                group_j = _choose_other_group(groups_list[cm], cdf=group_probs_i)
```

Comment:
In the docstring it says that the `group_probs_list` contains the probabilities (not cumulative) of meeting someone from another group.  Above in `calculate_infections` which calls this function `group_probs_list` is created from `group_probs` which are also according to the docstring not cumulative and not added up in the loop where you create the `group_prob_list` (86-88). I found that you return cumulative probabilities in `create_group_transition_probs` so I guess you only need to update the docstrings.

Reply:
This has to be updated.